Amplify/Categories/API/ClientBehavior/APICategoryGraphQLBehavior.swift

```diff
@@ -39,10 +39,12 @@ public protocol APICategoryGraphQLBehavior: class {
     ///
     /// - Parameters:
     ///   - model: The instance of the `Model`.
+    ///   - predicate: the condition that specifies whether the request should succeed or not
     ///   - type: The type of mutation to apply on the instance of `Model`.
     ///   - listener: The event listener for the operation
     /// - Returns: The AmplifyOperation being enqueued.
     func mutate<M: Model>(of model: M,
+                          where predicate: QueryPredicate?,
```

Comment:
In some of the unit tests, we are passing in nil --maybe it would be easier for customers to default this parameter to nil?

Reply:
> In some of the unit tests, we are passing in nil --maybe it would be easier for customers to default this parameter to nil?

We'd have to declare defaults in the implementation; protocol declarations can't have default values assigned in method signatures